sample-operators/tomcat-operator/src/main/java/io/javaoperatorsdk/operator/sample/WebappReconciler.java

```diff
@@ -70,13 +71,13 @@ public WebappReconciler(KubernetesClient kubernetesClient) {
       return UpdateControl.noUpdate();
     }
 
-    Tomcat tomcat = context.getSecondaryResource(Tomcat.class);
-    if (tomcat == null) {
+    Optional<Tomcat> tomcatValue = context.getSecondaryResource(Tomcat.class);
+    if (tomcatValue.isEmpty()) {
       throw new IllegalStateException("Cannot find Tomcat " + webapp.getSpec().getTomcat()
           + " for Webapp " + webapp.getMetadata().getName() + " in namespace "
           + webapp.getMetadata().getNamespace());
     }
-
+    Tomcat tomcat = tomcatValue.get();
```

Comment:
This makes the usage a little bit more complex, though I guess it forces users to handle the `null` case. This is something I often wonder about actually when designing an API: should we use `Optional` in similar cases or follow a `Map`-like convention of returning `null` if the element is not present?

Reply:
IMHO, map like convention was there before `Optional` was present in Java. It make sense and happens in multiple examples that the secondary resources is checked for null. It should be, since it's not necessary present, if not created yet for example. For that case it's better to use `Optional` I think. Exactly as you say, that it forces users to check, also to eliminate `null` checks